homeassistant/components/mqtt/__init__.py

```diff
@@ -1050,18 +1053,14 @@ def _raise_on_error(result_code: int) -> None:
         )
 
 
-def _match_topic(subscription: str, topic: str) -> bool:
-    """Test if topic matches subscription."""
+def _matcher_for_topic(topic: str) -> Any:
     # pylint: disable=import-outside-toplevel
     from paho.mqtt.matcher import MQTTMatcher
 
     matcher = MQTTMatcher()
-    matcher[subscription] = True
-    try:
-        next(matcher.iter_match(topic))
-        return True
-    except StopIteration:
-        return False
+    matcher[topic] = True
```

Comment:
To avoid re-using var name

```suggestion
    matcher[subscription] = True
```